domain/src/test/java/org/oppia/android/domain/question/QuestionAssessmentProgressControllerTest.kt

```diff
@@ -1019,6 +1028,368 @@ class QuestionAssessmentProgressControllerTest {
+    submitNumericInputAnswerAndMoveToNextQuestion(4.0)
+    verify(
+      mockCurrentQuestionLiveDataObserver,
+      atLeastOnce()
+    ).onChanged(currentQuestionResultCaptor.capture())
+    val ephemeralQuestion = currentQuestionResultCaptor.value.getOrThrow()
+    assertThat(ephemeralQuestion.ephemeralState.pendingState.wrongAnswerCount)
+      .isEqualTo(5)
+    val hintAndSolution = ephemeralQuestion.ephemeralState.state.interaction.getHint(0)
+    assertThat(hintAndSolution.hintContent.html)
+      .contains("<p>Hint text will appear here</p>")
+    // view question 2 hint
+    questionAssessmentProgressController.submitHintIsRevealed(
+      ephemeralQuestion.ephemeralState.state, true, 0
+    )
+    // submit correct answer from hint
+    submitNumericInputAnswerAndMoveToNextQuestion(3.0)
+
+    // submit correct answers for questions 5 and 4
+    submitNumericInputAnswerAndMoveToNextQuestion(5.0)
+    submitMultipleChoiceAnswerAndMoveToNextQuestion(1)
+
+    // check the computed score
+    subscribeToScoreCalculations(TEST_SKILL_ID_LIST_2)
+    testCoroutineDispatchers.runCurrent()
+    verify(
+      mockScoreCalculationsLiveDataObserver,
+      atLeastOnce()
+    ).onChanged(scoreCalculationCaptor.capture())
+
+    val userAssessmentPerformance = scoreCalculationCaptor.value.getOrThrow()
+    val grade = FractionGrade.newBuilder().apply {
+      numerator = 2.4
+      denominator = 3.0
+    }.build()
+
+    assertThat(userAssessmentPerformance.totalFractionScore).isEqualTo(grade)
+    assertThat(userAssessmentPerformance.fractionScorePerSkillMappingCount).isEqualTo(1)
+    assertThat(userAssessmentPerformance.getFractionScorePerSkillMappingOrThrow(TEST_SKILL_ID_2))
+      .isEqualTo(grade)
+  }
+
+  @Test
+  fun noHints_noWrongAnswers_noSolutionsViewed_returnPerfectScore() {
+    setUpTestApplicationWithSeed(questionSeed = 0)
+    subscribeToCurrentQuestionToAllowSessionToLoad()
+    startTrainingSession(TEST_SKILL_ID_LIST_2)
+
+    // submit correct answers for all questions (questions 2, 5, 4)
+    submitNumericInputAnswerAndMoveToNextQuestion(3.0)
+    submitNumericInputAnswerAndMoveToNextQuestion(5.0)
+    submitMultipleChoiceAnswerAndMoveToNextQuestion(1)
+
+    // check the computed score
+    subscribeToScoreCalculations(TEST_SKILL_ID_LIST_2)
+    testCoroutineDispatchers.runCurrent()
+    verify(
+      mockScoreCalculationsLiveDataObserver,
+      atLeastOnce()
+    ).onChanged(scoreCalculationCaptor.capture())
+
+    val userAssessmentPerformance = scoreCalculationCaptor.value.getOrThrow()
+    val grade = FractionGrade.newBuilder().apply {
+      numerator = 3.0
+      denominator = 3.0
+    }.build()
+
+    assertThat(userAssessmentPerformance.totalFractionScore).isEqualTo(grade)
+    assertThat(userAssessmentPerformance.fractionScorePerSkillMappingCount).isEqualTo(1)
+    assertThat(userAssessmentPerformance.getFractionScorePerSkillMappingOrThrow(TEST_SKILL_ID_2))
+      .isEqualTo(grade)
+  }
+
+  @Test
```

Comment:
Can you add a test where hints are viewed for multiple questions? So one hint and then the right answer , and repeat this for two questions 
Another one is to test two hints for a single question 
Otherwise I think the tests capture most use cases

Reply:
Added tests